shared/src/conversions.rs

```diff
@@ -28,6 +26,9 @@ pub fn u256_to_big_rational(input: &U256) -> BigRational {
 }
 
 pub fn big_int_to_u256(input: &BigInt) -> Result<U256> {
+    if input.is_zero() {
```

Comment:
This is unrelated to the change, but I found this issue where `big_int_to_u256(0)` would error during development. Since @fedgiac introduced the fixed point type for the weight, I ended up no longer needing this, but I left the fix in. If y'all feel strongly about it I can make a mini PR with just this change.